cpp/src/arrow/io/io-file-test.cc

```diff
@@ -41,14 +41,34 @@ static bool FileExists(const std::string& path) {
   return std::ifstream(path.c_str()).good();
 }
 
+#if defined(_MSC_VER)
+void customInvalidParameterHandler(const wchar_t* expression,
+	const wchar_t* function,
+	const wchar_t* file,
+	unsigned int line,
+	uintptr_t pReserved)
+{
+	wprintf(L"Invalid parameter detected in function %s."
+		L" File: %s Line: %d\n", function, file, line);
+	wprintf(L"Expression: %s\n", expression);
+}
```

Comment:
Was this code copy and pasted from https://msdn.microsoft.com/en-us/library/a9yf33zb.aspx? If so, what is the license? It looks like this is MS Public License, but we need to be sure. In general, if any code is contributed to an ASF project, it must be your original work, otherwise it must be attributed to a source with a compatible license. I might suggest writing a new function here